ksql-rest-app/src/test/java/io/confluent/ksql/rest/integration/AuthorizationFunctionalTest.java

```diff
@@ -0,0 +1,210 @@
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+@Category({IntegrationTest.class})
+public class AuthorizationFunctionalTest {
+  private static final IntegrationTestHarness TEST_HARNESS = IntegrationTestHarness.build();
+
+  private static final TestKsqlRestApp REST_APP = TestKsqlRestApp
+      .builder(TEST_HARNESS::kafkaBootstrapServers)
+      .withProperty("authentication.method", "BASIC")
+      .withProperty("authentication.roles", "**")
+      // Reuse the Kafka JAAS config for KSQL authentication which has the same valid users
+      .withProperty("authentication.realm", JAAS_KAFKA_PROPS_NAME)
+      .withProperty(
+          KsqlConfig.KSQL_SECURITY_EXTENSION_CLASS,
+          MockKsqlSecurityExtension.class.getName()
+      )
+      .withServiceContextBinder(config -> new AbstractBinder() {
+        @Override
+        protected void configure() {
+          bindFactory(new Factory<ServiceContext>() {
+            @Override
+            public ServiceContext provide() {
+              return TEST_HARNESS.getServiceContext();
+            }
+
+            @Override
+            public void dispose(final ServiceContext serviceContext) {
+              // do nothing because TEST_HARNESS#getServiceContext always
+              // returns the same instance
+            }
+          })
+              .to(ServiceContext.class)
+              .in(RequestScoped.class);
+        }
+      })
+      .build();
+
+  private static Set<String> allowedUsers = new HashSet<>();
+
+  @Mock
+  private KsqlAuthorizationProvider authorizationProvider;
+
+  @ClassRule
+  public static final RuleChain CHAIN = RuleChain.outerRule(TEST_HARNESS).around(REST_APP);
+
+  @Rule
+  public final ExpectedException expectedException = ExpectedException.none();
+
+  @Before
+  public void setUp() {
+    allowedUsers.clear();
+    MockKsqlSecurityExtension.setAuthorizationProvider(authorizationProvider);
+  }
+
+  @Test
+  public void shouldLoadSecurityExtension() {
+    // Then:
+    assertThat(MockKsqlSecurityExtension.initialized, is(true));
+    assertThat(MockKsqlSecurityExtension.closed, is(false));
+  }
+
+  @Test
+  public void shouldDenyAccess() {
+    // Then:
+    expectedException.expect(AssertionError.class);
+    expectedException.expectMessage(
+        String.format("Access denied to User:%s", VALID_USER1.username)
+    );
+
+    // When:
+    makeKsqlRequest(VALID_USER1, "SHOW TOPICS;");
+  }
+
+  @Test
+  public void shouldAllowAccess() {
+    // Given:
+    givenAuthorizedUser(VALID_USER1);
+
+    // When:
+    final List<KsqlEntity> results = makeKsqlRequest(VALID_USER1, "SHOW TOPICS;");
+
+    // Then:
+    assertSuccessful(results);
+  }
+
+  private void givenAuthorizedUser(final Credentials user) {
+    allowedUsers.add(user.username);
+  }
+
+  private static void assertSuccessful(final List<KsqlEntity> results) {
+    results.stream()
```

Comment:
Is this check doing anything? I don't think `SHOW TOPICS;` returns a CommandStatusEntity. By having the check filter for CommandStatusEntities and only perform checks on the results that match, the test could pass without performing any checks. It would be better to directly assert what we expect about the results, for example, that results.size() is 1, and that one entity is of type `KafkaTopicsList`, etc.

Reply:
I copied this method from another integration test. I didn't notice the results weren't correct.
I changed it to verify a topic was returned.